data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -307,6 +307,13 @@ private Map<String, String> generateIngestAndCommandHeaders(ClientRequestPropert
                 activityId, activityId, activityId, clientRequestId);
         headers.put("x-ms-activitycontext", activityContext);
 
+        // replace non-ascii characters in header values with '?'
+        for (Map.Entry<String, String> entry : headers.entrySet()) {
```

Comment:
nit: I like how you did it as well, but I think this is a great scenario for using the stream API.

Reply:
Done